master/buildbot/www/rest.py

```diff
@@ -57,12 +56,13 @@ def __init__(self, message, jsonrpccode):
 
 
 class ContentTypeParser:
-    def __init__(self, contenttype):
+    def __init__(self, contenttype: str | bytes | None) -> None:
         self.typeheader = contenttype
 
-    def gettype(self):
-        mimetype, _ = cgi.parse_header(bytes2unicode(self.typeheader))
-        return mimetype
+    def gettype(self) -> str | None:
+        if self.typeheader is None:
+            return None
+        return bytes2unicode(self.typeheader).split(';', 1)[0]
```

Comment:
The CGI.parse_header doc states "Parse a MIME header (such as Content-Type)", so I assumed it might be an other header than Content-Type that need this.

In any case, this code if to parse the 'parameters' (such as charset and boundary), and since Buildbot only need the media type, we can ignore it.